spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/partitioner/ReplicaPartitioner.scala

```diff
@@ -17,7 +20,7 @@ class ReplicaPartitioner(partitionsPerReplicaSet: Int, connector: CassandraConne
   /* TODO We Need JAVA-312 to get sets of replicas instead of single endpoints. Once we have that we'll be able to
   build a map of Set[ip,ip,...] => Index before looking at our data and give the all options for the preferred location
    for a partition*/
-  private val hosts = connector.hosts.toVector
+  private val hosts = connector.withClusterDo(_.getMetadata.getAllHosts).map(_.getAddress).toVector
```

Comment:
Sounds good to me, I also realized that this implementation may have some issues on multi-dc clusters. So I think I need to make a slight change around that as well